compiler/rustc_type_ir/src/visit.rs

```diff
@@ -326,6 +326,12 @@ pub trait TypeVisitableExt<I: Interner>: TypeVisitable<I> {
         self.has_type_flags(TypeFlags::HAS_FREE_REGIONS)
     }
 
+    fn has_regions(&self) -> bool {
+        self.has_type_flags(
+            TypeFlags::HAS_FREE_REGIONS | TypeFlags::HAS_RE_ERASED | TypeFlags::HAS_RE_BOUND,
```

Comment:
:thinking: we should support bound regions and not check for them

maybe `has_free_or_erased_regions`... or actually, `has_free_regions` should include `RE_ERASED` imo, so we should have `has_free_non_erased_regions` :<

Reply:
Hmm yeah ur right there's no real reason to forbid bound regions. I was being a bit overly conservative here :3 I just removed `has_regions` entirely because if its only FREE_REGIONS | ERASED then its not that bad to just do it in-line in args lowering.

It is somewhat weird that has_free_regions doesnt proc on RE_ERASED :thinking: